encode_test.go

```diff
@@ -713,26 +672,18 @@ func (ft *failingMarshaler) MarshalYAML() (interface{}, error) {
 
 func TestMarshalerError(t *testing.T) {
 	_, err := yaml.Marshal(&failingMarshaler{})
-	if err != failingErr {
-		t.Fatalf("Marshal() returned %v, want %v", err, failingErr)
-	}
+	assert.Equal(t, err, failingErr)
```

Comment:
Error should be tested with ErrorIs and not Equal.

Error could be wrapped, and here it would fail.

So adding a `ErrorIs` asserter could be a good option.

We can leave it like this, but I prefer to raise the point.

And, yes, I know you are only refactoring, but I would prefer if this could be fixed

Reply:
What do you think about this?

Reply:
I will look into making the assertion after I get the argument order resolved :hourglass_flowing_sand:

Reply:
done, please take a look